sound/soc/sof/intel/bdw.c

```diff
@@ -640,24 +640,13 @@ static int bdw_probe(struct snd_sof_dev *sdev)
 	return ret;
 }
 
-#define BDW_FORMATS (SNDRV_PCM_FMTBIT_S16_LE | SNDRV_PCM_FMTBIT_S24_LE | \
-	SNDRV_PCM_FMTBIT_S32_LE)
-
```

Comment:
these formats may not be used, but at the same time we currently have restrictions on what the topology can do (single format per DAI), and there will be no check that the topology settings are correct and supported by the firmware. We should probably have a layer that does check the topology is indeed supported, otherwise with a single error in a topology file we have a risk of reaching unsupported configurations. it's just too risky.
In short, I would really like the topology to support multiple DAI configs/formats, and a per-platform check in the driver.

Reply:
@plbossart makes sense. I can add the check but looking at the formats defined for all platforms, I see no restrictions. All platforms seem to support 16/24/32 bit formats. Is that a mistake? If it is, could you please remind me what they should look like?